pallets/ajuna-tournament/src/lib.rs

```diff
@@ -0,0 +1,753 @@
+				.ok_or::<Error<T, I>>(Error::<T, I>::TournamentNotFound)?;
+
+			TournamentRankings::<T, I>::mutate(season_id, tournament_id, |table| {
+				if let Err(index) = table.binary_search_by(|(other_id, other)| {
+					ranker.rank_against((entity_id, entity), (other_id, other))
+				}) {
+					match Self::try_update_rank_table(
+						table,
+						&tournament_config,
+						index,
+						entity_id,
+						entity,
+					) {
+						None => Ok(()),
+						Some(true) => {
+							Self::deposit_event(
+								crate::pallet::Event::<T, I>::EntityEnteredRanking {
+									season_id: *season_id,
+									tournament_id,
+									entity_id: entity_id.clone(),
+								},
+							);
+							Ok(())
+						},
+						Some(false) => Err(Error::<T, I>::FailedToRankEntity.into()),
+					}
+				} else {
+					Ok(())
+				}
+			})
+		}
+
+		fn try_rank_entity_for_golden_duck(
+			season_id: &T::SeasonId,
+			entity_id: &T::EntityId,
+		) -> DispatchResult {
+			let tournament_id = Self::try_get_active_tournament_id_for(season_id)?;
+
+			GoldenDucks::<T, I>::mutate(season_id, tournament_id, |state| match state {
+				GoldenDuckState::Enabled(payout_perc, None) => {
+					*state = GoldenDuckState::Enabled(*payout_perc, Some(entity_id.clone()));
+
+					Self::deposit_event(crate::pallet::Event::<T, I>::EntityBecameGoldenDuck {
+						season_id: *season_id,
+						tournament_id,
+						entity_id: entity_id.clone(),
+					});
+				},
+				GoldenDuckState::Enabled(payout_perc, Some(ref entry_id))
+					if entity_id < entry_id =>
+				{
+					*state = GoldenDuckState::Enabled(*payout_perc, Some(entity_id.clone()));
+
+					Self::deposit_event(crate::pallet::Event::<T, I>::EntityBecameGoldenDuck {
+						season_id: *season_id,
+						tournament_id,
+						entity_id: entity_id.clone(),
+					});
+				},
+				_ => {},
+			});
+
+			Ok(())
+		}
+	}
+
+	impl<T: Config<I>, I: 'static> TournamentClaimer<T::SeasonId, AccountIdFor<T>, T::EntityId>
+		for Pallet<T, I>
+	{
+		fn try_claim_tournament_reward_for(
+			season_id: &T::SeasonId,
+			account: &AccountIdFor<T>,
+			entity_id: &T::EntityId,
+		) -> DispatchResult {
+			match ActiveTournaments::<T, I>::get(season_id) {
+				TournamentState::Inactive => Err(Error::<T, I>::NoActiveTournamentForSeason.into()),
+				TournamentState::ActivePeriod(_) =>
+					Err(Error::<T, I>::TournamentNotInClaimPeriod.into()),
+				TournamentState::ClaimPeriod(tournament_id, reward_pot) => {
+					if let Some(index) = TournamentRankings::<T, I>::get(season_id, tournament_id)
+						.iter()
+						.position(|(entry_id, _)| entry_id == entity_id)
+					{
+						let tournament_config = Tournaments::<T, I>::get(season_id, tournament_id)
+							.ok_or(Error::<T, I>::TournamentNotFound)?;
+						let treasury_account = Self::tournament_treasury_account_id(*season_id);
+
+						let payout_percentage = tournament_config
+							.reward_distribution
+							.into_iter()
+							.nth(index)
+							.unwrap_or_default();
```

Comment:
I see you are trying to remove the need for copy, but maybe `reward_distribution.get(index).unwrap_or_default()`, followed by a deref op is better? Not sure, just giving ideas here.